examples/data_parallel_retrieval.py

```diff
@@ -0,0 +1,230 @@
+"""
+# Recommending movies: retrieval with data parallel training
+
+In this tutorial, we are going to train the exact same retrieval model as we
+did in our
+[basic retrieval](https://github.com/keras-team/keras-rs/blob/main/examples/basic_retrieval.py)
+tutorial, but in a distributed way.
+
+Distributed training is used to train models on multiple devices or machines
+simultaneously, thereby reducing training time. Here, we focus on synchronous
+data parallel training. Each accelerator (GPU/TPU) holds a complete replica
+of the model, and sees a different mini-batch of the input data. Local gradients
+are computed on each device, aggregated and used to compute a global gradient
+update.
+
+Before we begin, let's note down a few things:
+
+1. The number of accelerators should be greater than 1.
+2. The `keras.distribution` API works only with JAX. So, make sure you select
+   JAX as your backend!
+"""
+
+import os
+
+os.environ["KERAS_BACKEND"] = "jax"
+
+import random
+
+import jax
+import keras
+import tensorflow as tf  # Needed only for the dataset
+import tensorflow_datasets as tfds
+
+import keras_rs
+
+"""
+## Data Parallel
+
+For the synchronous data parallelism strategy in distributed training,
+we will use the `DataParallel` class present in the `keras.distribution`
+API.
+"""
+devices = jax.devices("gpu")  # Assume it has >1 local GPUs.
+data_parallel = keras.distribution.DataParallel(devices=devices)
+
+"""
+Alternatively, you can choose to create the `DataParallel` object
+using a 1D `DeviceMesh` object, like so:
+
+```
+mesh_1d = keras.distribution.DeviceMesh(
+    shape=(8,), axis_names=["data"], devices=devices
```

Comment:
What is `8` here? Isn't it the number of devices? Can we put `len(devices)` instead?

Reply:
Oh, yeah. Missed this.